libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -993,68 +1108,77 @@ void AC_PosControl::update_U_controller()
 
-    _accel_target_neu_cmss.z = _pid_vel_u.update_all(_vel_target_neu_cms.z, _vel_estimate_neu_cms.z, _dt_s, _motors.limit.throttle_lower, _motors.limit.throttle_upper);
-    _accel_target_neu_cmss.z *= AP::ahrs().getControlScaleZ();
+    // PID controller: convert velocity error to acceleration
+    _accel_target_neu_mss.z = _pid_vel_u.update_all(_vel_target_neu_ms.z, _vel_estimate_neu_ms.z, _dt_s, _motors.limit.throttle_lower, _motors.limit.throttle_upper);
+    _accel_target_neu_mss.z *= AP::ahrs().getControlScaleZ();
 
     // add feed forward component
-    _accel_target_neu_cmss.z += _accel_desired_neu_cmss.z + _accel_offset_neu_cmss.z + _accel_terrain_u_cmss;
+    _accel_target_neu_mss.z += _accel_desired_neu_mss.z + _accel_offset_neu_mss.z + _accel_terrain_u_mss;
 
     // Acceleration Controller
 
-    // Calculate vertical acceleration
-    const float measured_accel_u_cmss = get_measured_accel_U_cmss();
+    // Gravity-compensated vertical acceleration measurement (positive = up)
+    const float measured_accel_u_mss = get_measured_accel_U_mss();
 
-    // ensure imax is always large enough to overpower hover throttle
-    if (_motors.get_throttle_hover() * 1000.0f > _pid_accel_u.imax()) {
-        _pid_accel_u.set_imax(_motors.get_throttle_hover() * 1000.0f);
+    // Ensure integrator can produce enough thrust to overcome hover throttle
+    if (_motors.get_throttle_hover() * 10.0 * 100.0 > _pid_accel_u.imax()) {
+        _pid_accel_u.set_imax(_motors.get_throttle_hover() * 10.0 * 100.0);
     }
     float thr_out;
     if (_vibe_comp_enabled) {
+        // Use vibration-resistant throttle estimator (feedforward + scaled integrator)
         thr_out = get_throttle_with_vibration_override();
     } else {
-        thr_out = _pid_accel_u.update_all(_accel_target_neu_cmss.z, measured_accel_u_cmss, _dt_s, (_motors.limit.throttle_lower || _motors.limit.throttle_upper)) * 0.001f;
-        thr_out += _pid_accel_u.get_ff() * 0.001f;
+        // Standard PID update using vertical acceleration error
+        thr_out = _pid_accel_u.update_all(_accel_target_neu_mss.z * 100.0, measured_accel_u_mss * 100.0, _dt_s, (_motors.limit.throttle_lower || _motors.limit.throttle_upper)) * 0.1 * 0.01;
+        // Include FF contribution to reduce delay
+        thr_out += _pid_accel_u.get_ff() * 0.1 * 0.01;
     }
     thr_out += _motors.get_throttle_hover();
 
     // Actuator commands
 
-    // send throttle to attitude controller with angle boost
+    // Send final throttle output to attitude controller (includes angle boost)
     _attitude_control.set_throttle_out(thr_out, true, POSCONTROL_THROTTLE_CUTOFF_FREQ_HZ);
 
     // Check for vertical controller health
 
-    // _speed_down_cms is checked to be non-zero when set
-    float error_ratio = _pid_vel_u.get_error() / _vel_max_down_cms;
+    // Update health indicator based on error magnitude vs configured speed range
+    float error_ratio = _pid_vel_u.get_error() / _vel_max_down_ms;
```

Comment:
the vertical velocity PID's error has been changed to m/s right?  So only the horizontal velocity PID error remains in cms?